backend/dgraph/src/upsert_entity.rs

```diff
@@ -0,0 +1,179 @@
+use gql_client::GraphQLError;
+use serde::{Deserialize, Serialize};
+
+use crate::{DgraphClient, EntityInput};
+
+#[derive(Serialize, Debug)]
+struct UpsertVars {
+    input: EntityInput,
+    upsert: bool,
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Debug, Clone)]
+pub struct UpsertResponseData {
+    pub data: UpsertResponse,
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Debug, Clone)]
+pub struct UpsertResponse {
+    pub numUids: u32,
```

Comment:
Since we don't know what type of mutation we're going to do, we can't return the entity data as part of the query :(